src/rust/engine/src/intrinsics/digests.rs

```diff
@@ -354,3 +361,153 @@ fn digest_subset_to_digest(digest_subset: Value) -> PyGeneratorResponseNativeCal
+                            .set_item("length", m.length.to_object(py))
+                            .expect("set length for PathMetadata");
+
+                        kwargs
+                            .set_item("is_executable", m.is_executable.to_object(py))
+                            .expect("set is_executable for PathMetadata");
+
+                        kwargs
+                            .set_item("read_only", m.read_only.to_object(py))
+                            .expect("set read_only for PathMetadata");
+
+                        match m.unix_mode {
+                            Some(unix_mode) => kwargs
+                                .set_item("unix_mode", unix_mode.to_object(py))
+                                .expect("set unix_mode for PathMetadata"),
+                            None => kwargs
+                                .set_item("unix_mode", py.None())
+                                .expect("set unix_mode for PathMetadata"),
+                        }
+
+                        kwargs
+                            .set_item("accessed_time", py.None())
+                            .expect("set accessed_time for PathMetadata");
+
+                        kwargs
+                            .set_item("created_time", py.None())
+                            .expect("set created_time for PathMetadata");
+
+                        fn set_datetime_opt(
+                            py: Python<'_>,
+                            kwargs: &PyDict,
+                            key: &'static str,
+                            dt_opt: Option<SystemTime>,
+                        ) -> Result<(), PyErr> {
+                            match dt_opt {
+                                Some(st) => {
+                                    let duration_since_unix_epoch = st
+                                        .duration_since(UNIX_EPOCH)
+                                        .expect("duration since unix epoch");
+
+                                    let dt = chrono::DateTime::from_timestamp(
+                                        duration_since_unix_epoch.as_secs() as i64,
+                                        duration_since_unix_epoch.subsec_nanos(),
+                                    )
+                                    .unwrap();
+                                    let d = dt.date_naive();
+                                    let t = dt.time();
+
+                                    let py_datetime = PyDateTime::new_bound(
+                                        py,
+                                        d.year(),
+                                        d.month() as u8,
+                                        d.day() as u8,
+                                        t.hour() as u8,
+                                        t.minute() as u8,
+                                        t.second() as u8,
+                                        t.nanosecond() / 1000,
+                                        None,
```

Comment:
Since we're converting from a `SystemTime` which is a reference to an 'absolute' time since unix epoch, I think we theoretically know a timezone here (likely UTC) and dropping that may lead to incorrect/misleading results.

My computer is in UTC+1000. Running this code at 16:57 prints `2024-06-05 06:57:52.656043` i.e. the UTC timestamp of the current moment, not my local time:

```rust
use std::time::SystemTime;

fn main() {
    let st = SystemTime::now();

    let duration_since_unix_epoch = st
        .duration_since(SystemTime::UNIX_EPOCH)
        .expect("duration since unix epoch");
    let dt = chrono::DateTime::from_timestamp(
        duration_since_unix_epoch.as_secs() as i64,
        duration_since_unix_epoch.subsec_nanos(),
    )
    .unwrap();
    let d = dt.date_naive();
    let t = dt.time();

    println!("{d:?} {t:?}");
}
```

(Setup: `cargo new test-time; cd test-time; cargo add chrono`, then past that into `src/main.rs`, then `cargo run`)

I'm not exactly sure of the best way to handle that... and it might become irrelevant with the comment about pyo3 doing the conversion?

Reply:
I can just add the UTC time zone to the `PyDateTime` (which maps to `datetime.datetime`. My initial thought had been to go with the ["naive" version of `datetime` ](https://docs.python.org/3/library/datetime.html#aware-and-naive-objects), but you are correct: We know it is in UTC here so the logic should just configure that.